src/Toml/PrefixTree.hs

```diff
@@ -42,7 +44,13 @@ import qualified Data.Text as Text
 
 -- | Represents the key piece of some layer.
 newtype Piece = Piece { unPiece :: Text }
-    deriving (Show, Eq, Ord, Hashable, IsString)
+    deriving (Show, Eq, Ord, Hashable, Generic)
```

Comment:
I think here you can use `DerivingStrategies` extension to not write instances manually. This is a typical problem with `DeriveAnyClass` and `GeneralizedNewtypeDeriving` because GHC doesn't know what strategy are you using. But since GHC 8.2.2 you can specify strategy explicitly.

```suggestion
    deriving stock (Generic)
    deriving newtype (Show, Eq, Ord, Hashable, IsString, NFData)
```